src/frauddetectionservice/src/main/kotlin/frauddetectionservice/main.kt

```diff
@@ -46,8 +58,27 @@ fun main() {
                     val newCount = accumulator + 1
                     val orders = OrderResult.parseFrom(record.value())
                     logger.info("Consumed record with orderId: ${orders.orderId}, and updated total count to: $newCount")
+                    if (getFeatureFlagValue("kafkaQueueProblems") > 0) {
+                        logger.info("FeatureFlag 'kafkaQueueProblems' is enabled, sleeping 1 second")
+                        Thread.sleep(1000)
+                    }
                     newCount
                 }
         }
     }
 }
+
+/**
+* Retrieves the status of a feature flag from the Feature Flag service.
+*
+* @param ff The name of the feature flag to retrieve.
+* @return `true` if the feature flag is enabled, `false` otherwise or in case of errors.
+*/
+fun getFeatureFlagValue(ff: String): Int {
+    val client = OpenFeatureAPI.getInstance().client
+    // TODO: Plumb the actual session ID from the frontend via baggage?
+    val uuid = UUID.randomUUID()
+    client.evaluationContext = MutableContext().add("session", uuid.toString())
```

Comment:
I recommend using `ImmutableContext` and setting the session id in the constructor. That will be mapped to the `targetKey` property in the evaluation context, which is used in the [pseudo-random bucketing algorithm](https://flagd.dev/reference/custom-operations/fractional-operation/).

Reply:
done